packages/amplication-git-service/src/git-client-factory.module.ts

```diff
@@ -0,0 +1,17 @@
+import { Global, Module } from '@nestjs/common';
+import { ConfigModule } from '@nestjs/config';
+import { GithubFactory } from './providers/github.factory';
+
+@Global()
+@Module({
+  imports: [
+    ConfigModule,
+    ConfigModule.forRoot({
+      isGlobal: true,
+      envFilePath: ['.env.local', '.env']
+    })
+  ],
+  providers: [GithubFactory],
+  exports: [GithubFactory]
+})
+export class GitClientFactoryModule {}
```

Comment:
Can you explain the usage of `Factory` here? NestJS modules are stateful, where does the factory pattern play here?

If this module's purpose is to be a factory that generates instances, then this should not be a module, but rather a an exported class/function from a non-NestJS file.

Factory-generated instances won't have access to the module's DI state anyway, so it's kind of pointless.

Reply:
This factory is made for a transaction-like functionality (It's basically syntactic sugar to a closure that holds state). will love to see the nest factory that you mentioned it may be a good fit here. if you have a few minutes to spare lets have a quick call on it.